### PR TITLE
Advanced search opportunities for mass edit

### PR DESCRIPTION
**Pull request by teolemon**
Advanced search opportunities for mass edit

### CODE REVIEW

File: OpenFoodFactsPower.user.js [hunk 1]
```diff
@@ -523,7 +524,16 @@ content: " — ";
         pageType === "product view"||
         pageType === "saved-product page") {
 
-
+	// Showing it directly on the product page, for emerging categories. 
+	// https://world.openfoodfacts.org/cgi/search.pl?action=process&sort_by=unique_scans_n&page_size=20&action=display&tagtype_0=states&tag_contains_0=contains&tag_0=categories%20to%20be%20completed&search_terms=lasagne
+        var SearchUncategorizedProductsOpportunitiesDeepLink = normalizeTagName($("h1[itemprop='name']").text())
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:529]:
```suggestion
        var SearchUncategorizedProductsOpportunitiesDeepLink = normalizeTagName($("h1[itemprop='name']").text());
```

File: OpenFoodFactsPower.user.js [hunk 2]
```diff
@@ -523,7 +524,16 @@ content: " — ";
...
+            : "");
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:530]:
```suggestion

```

File: OpenFoodFactsPower.user.js [hunk 3]
```diff
@@ -523,7 +524,16 @@ content: " — ";
...
+        $("#SearchUncategorizedProductsOpportunities").after(
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:531]:
```suggestion
        $("#hungerGameLink").after(
```

File: OpenFoodFactsPower.user.js [hunk 4]
```diff
@@ -523,7 +524,16 @@ content: " — ";
...
+            ((SearchUncategorizedProductsOpportunitiesDeepLink) ? '<p>'+
+            '> <a href="https://world.openfoodfacts.org/cgi/search.pl?action=process&sort_by=unique_scans_n&page_size=20&action=display&tagtype_0=states&tag_contains_0=contains&tag_0=categories%20to%20be%20completed&search_terms=' + SearchUncategorizedProductsOpportunitiesDeepLink + '">' +
+            'Categorization opportunities using Mass Edit</a>' +
+            '</p>' : "")
+
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:536]:
```suggestion

```

File: OpenFoodFactsPower.user.js [hunk 5]
```diff
@@ -523,7 +524,16 @@ content: " — ";
...
         pageType === "saved-product page") {
 
-
+	// Showing it directly on the product page, for emerging categories. 
+	// https://world.openfoodfacts.org/cgi/search.pl?action=process&sort_by=unique_scans_n&page_size=20&action=display&tagtype_0=states&tag_contains_0=contains&tag_0=categories%20to%20be%20completed&search_terms=lasagne
+        var SearchUncategorizedProductsOpportunitiesDeepLink = normalizeTagName($("h1[itemprop='name']").text())
+            : "");
+        $("#SearchUncategorizedProductsOpportunities").after(
+            ((SearchUncategorizedProductsOpportunitiesDeepLink) ? '<p>'+
+            '> <a href="https://world.openfoodfacts.org/cgi/search.pl?action=process&sort_by=unique_scans_n&page_size=20&action=display&tagtype_0=states&tag_contains_0=contains&tag_0=categories%20to%20be%20completed&search_terms=' + SearchUncategorizedProductsOpportunitiesDeepLink + '">' +
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:533]:
```suggestion
            '> <a href="/cgi/search.pl?action=process&sort_by=unique_scans_n&page_size=20&action=display&tagtype_0=states&tag_contains_0=contains&tag_0=categories%20to%20be%20completed&search_terms=' + SearchUncategorizedProductsOpportunitiesDeepLink + '">' +
```

File: OpenFoodFactsPower.user.js [hunk 6]
```diff
@@ -523,7 +524,16 @@ content: " — ";
...
+            'Categorization opportunities using Mass Edit</a>' +
+            '</p>' : "")
```
Comment (CharlesNepote) [inline OpenFoodFactsPower.user.js:535]:
```suggestion
            '</p>' : "");
```

**Comment by teolemon** [comment]
Those cases with brands, we won't be able to solve them immediately. There's already a positive impact for already clean product names.

**Pull request merged**